src/main/java/com/checkmarx/jenkins/tools/CheckmarxInstaller.java

```diff
@@ -195,6 +237,44 @@ public Void call() throws IOException {
             return null;
         }
 
+        public static void copyURLToFile(URL source, String proxyStr, String proxyType, File destination, int connectionTimeoutMillis, int readTimeoutMillis) throws IOException {
+            URLConnection connection;
+            if (proxyStr == null) {
+                connection = source.openConnection();
+            } else {
+                try {
+                    URI proxyUrl = new URI(proxyStr);
+                    Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyUrl.getHost(), proxyUrl.getPort()));
+                    connection = source.openConnection(proxy);
+                    if (StringUtils.isNotEmpty(proxyUrl.getUserInfo())) {
+                        // Proxy With UserInfo Not Checked !
+                        String authHeader = new String(Base64.getEncoder().encode(proxyUrl.getUserInfo().getBytes(UTF_8)), UTF_8).replace("\r\n", "");
+                        connection.setRequestProperty("Proxy-Authorization", getAuthProxyPrefix(proxyType) + authHeader);
+                        String[] userPass = proxyUrl.getUserInfo().split(":");
+                        Authenticator.setDefault(new MyAuthenticator(userPass[0], userPass[1]));
+                    }
+                } catch (Exception e) {
+                    throw new ToolDetectionException("failed to create proxy with " + proxyStr, e);
+                }
+            }
+
+            connection.setConnectTimeout(connectionTimeoutMillis);
+            connection.setReadTimeout(readTimeoutMillis);
+            connection.connect();
+            InputStream stream = connection.getInputStream();
+            try {
+                FileUtils.copyInputStreamToFile(stream, destination);
+            } catch (Throwable var15) {
```

Comment:
have a regular catch too? or it can throw only var15. ?

Reply:
Throwable catches all the exceptions.